server/src/main/java/org/apache/druid/server/SegmentManager.java

```diff
@@ -171,9 +167,16 @@ public boolean loadSegment(final DataSegment segment) throws SegmentLoadingExcep
           );
 
           if ((entry != null) && (entry.getChunk(segment.getShardSpec().getPartitionNum()) != null)) {
-            log.warn("Told to load a adapter for a segment[%s] that already exists", segment.getId());
+            log.warn("Told to load an adapter for segment[%s] that already exists", segment.getId());
             resultSupplier.set(false);
           } else {
+            final Segment adapter;
+            try {
+              adapter = getAdapter(segment);
+            }
+            catch (SegmentLoadingException e) {
```

Comment:
SegmentLoadingException is specifically checked for in a few places that treat it differently from a generic runtime exception. Please keep it.